File: backend/Walkers.Books/Walkers.Books.Api/Models/Book.cs
namespace Walkers.Books.Api.Models;

public class Book
{
    public Guid Id { get; set; }
    public string Title { get; init; }
    public string Author { get; init; }
    public string ISBN { get; init; }
    public int? Rating { get; set; }
    public List<string>? Comments { get; set; }
    public bool HasNotes => Comments is not null && Comments.Count > 0;
    public string? CoverImageUrl { get; init; }
}

File: frontend/src/types/book.ts
export interface Book {
  id: string;
  title: string;
  author: string;
  isbn: string;
  rating?: number;
  comments?: string[];
  coverImageUrl?: string;
  hasNotes: boolean;
}

export interface CreateBookRequest {
  title: string;
  author: string;
  isbn: string;
  rating?: number;
  comment?: string;
  coverImageUrl?: string;
}

export interface UpdateBookRequest {
  bookId: string;
  rating?: number;
  comment?: string;
}
